src/librustc_data_structures/lib.rs

```diff
@@ -117,3 +117,34 @@ impl<F: Fn()> Drop for OnDrop<F> {
 // See comments in src/librustc/lib.rs
 #[doc(hidden)]
 pub fn __noop_fix_for_27438() {}
+
+/// Used as a return value to signify a fatal error occurred. (It is also
```

Comment:
Also, it would be easier to refactor and move this around if it were contained in its own module (I know I know... micro-modules and whatnot, but I think this would be better for maintainability)